actionview/lib/action_view/railtie.rb

```diff
@@ -78,6 +78,13 @@ class Railtie < Rails::Engine # :nodoc:
       PartialRenderer.collection_cache = app.config.action_controller.cache_store
     end
 
+    initializer "action_view.button_to_generates_button_tag" do |app|
+      ActiveSupport.on_load(:action_view) do
+        generates_button_tag = app.config.action_view.delete(:button_to_generates_button_tag)
+        ActionView::Helpers::FormHelper.button_to_generates_button_tag = generates_button_tag
```

Comment:
Think this would be clearer as:

```suggestion
        ActionView::Helpers::FormHelper.button_to_generates_button_tag =
          app.config.action_view.delete(:button_to_generates_button_tag)
```